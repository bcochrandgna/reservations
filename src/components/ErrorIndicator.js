// @flow
import React from 'react';
import { View, Text } from 'react-native';
import { Icon } from 'react-native-elements';
import styles from '../styles/components/ErrorIndicator';
import { errorTextColor } from '../styles';

type Props = {
  error: string,
};

const ErrorIndicator = (props: Props) => {
  const { error } = props;
  return (
    <View style={styles.container}>
      <Icon name="error" color={errorTextColor} />
      <Text style={styles.errorText}>{error}</Text>
    </View>
  );
};

export default ErrorIndicator;
