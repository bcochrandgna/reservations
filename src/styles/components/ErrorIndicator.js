import { StyleSheet } from 'react-native';
import { standardPadding, errorTextColor, standardMargin } from '..';

export default StyleSheet.create({
  container: {
    padding: standardPadding,
    flexDirection: 'row',
    alignItems: 'center',
  },
  errorText: {
    marginLeft: standardMargin,
    color: errorTextColor,
  },
});
