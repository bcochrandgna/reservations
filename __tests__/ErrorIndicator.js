import React from 'react';
import { shallow } from 'enzyme';
import ErrorIndicator from '../src/components/ErrorIndicator';

describe('ErrorIndicator', () => {
  it('renders correctly with null props', () => {
    const wrapper = shallow(<ErrorIndicator />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const wrapper = shallow(<ErrorIndicator error="Test Error" />);
    expect(wrapper).toMatchSnapshot();
  });
});
