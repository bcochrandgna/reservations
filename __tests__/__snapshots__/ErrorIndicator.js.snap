// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIndicator renders correctly with null props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 5,
    }
  }
>
  <Themed.Icon
    color="#ff7200"
    name="error"
  />
  <Text
    style={
      Object {
        "color": "#ff7200",
        "marginLeft": 5,
      }
    }
  />
</View>
`;

exports[`ErrorIndicator renders correctly with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 5,
    }
  }
>
  <Themed.Icon
    color="#ff7200"
    name="error"
  />
  <Text
    style={
      Object {
        "color": "#ff7200",
        "marginLeft": 5,
      }
    }
  >
    Test Error
  </Text>
</View>
`;
